src/Service/RequestService.php

```diff
@@ -321,6 +321,73 @@ private function recursiveRequestQueryKey(array &$vars, string $name, string $na
 
     }//end recursiveRequestQueryKey()
 
+    /**
+     * Gets the schemas related to this endpoint.
+     *
+     * @return array All necessary info from the schemas related to this endpoint.
+     */
+    private function getAllowedSchemas(): array
+    {
+        $allowedSchemas = [
+            'id'        => [],
+            'name'      => [],
+            'reference' => [],
+        ];
+
+        if (isset($this->data['endpoint']) === true) {
+            foreach ($this->data['endpoint']->getEntities() as $entity) {
+                $allowedSchemas['id'][]        = $entity->getId()->toString();
+                $allowedSchemas['name'][]      = $entity->getName();
+                $allowedSchemas['reference'][] = $entity->getReference();
+            }
+        }
+
+        return $allowedSchemas;
+
+    }//end getAllowedSchemas()
+
+    /**
+     * This function checks if the requesting user has the needed scopes to access the requested endpoint.
+     *
+     * @param array $references Schema references which we checks scopes for.
+     *
+     * @return null|Response A 403 response if the requested user does not have the needed scopes.
+     */
+    private function checkUserScopes(array $references, string $type = 'schemas'): ?Response
+    {
+        $userHasScope  = false;
+        $scopes        = $this->getScopes();
+        $loopedSchemas = [];
+        foreach ($references as $reference) {
+            $schemaScope     = "$type.$reference.{$this->data['method']}";
+            $loopedSchemas[] = $schemaScope;
+            if (isset($scopes[$schemaScope]) === true) {
+                // If true the user is authorized.
+                return null;
+            }
+        }
+
+        // If the user doesn't have the normal scope and doesn't have the admin scope, return a 403 forbidden.
+        if ($userHasScope === false && isset($scopes["admin.{$this->data['method']}"]) === false) {
```

Comment:
you don't need $userHasScope because you never set it to be true and just return null before this if